src/main/java/nextstep/fp/StreamStudy.java

```diff
@@ -5,40 +5,62 @@
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class StreamStudy {
 
-    public static long countWords() throws IOException {
-        String contents = new String(Files.readAllBytes(Paths
-                .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
-        List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
+    public static final String FILE_PATH = "src/main/resources/fp/war-and-peace.txt";
+    public static final String SPLIT_PATTERN = "[\\P{L}]+";
 
-        long count = 0;
-        for (String w : words) {
-            if (w.length() > 12) count++;
-        }
-        return count;
+    public static long countWords() throws IOException {
+        List<String> words = splitByPatten(readFromFile(), SPLIT_PATTERN);
+        return words.stream()
+                .filter(word -> word.length() > 12)
+                .count();
     }
 
     public static void printLongestWordTop100() throws IOException {
-        String contents = new String(Files.readAllBytes(Paths
-                .get("src/main/resources/fp/war-and-peace.txt")), StandardCharsets.UTF_8);
-        List<String> words = Arrays.asList(contents.split("[\\P{L}]+"));
-
-        // TODO 이 부분에 구현한다.
+        List<String> words = splitByPatten(readFromFile(), SPLIT_PATTERN);
+        words.stream()
+                .filter(word -> word.length() > 12)
+                .sorted(Comparator.comparing(String::length).reversed())
+                .limit(100)
+                .distinct()
+                .forEach(word -> System.out.println(word.toLowerCase()));
```

Comment:
똑같긴 하지만, 분리를 위해 아래와 같이도 사용하실 수 있을 것 같네요!

```suggestion
                .distinct()
                .map(String::toLowerCase)
                .forEach(System.out::println);
```

Reply:
훨씬 더 보기에 깔끔하네요 👍